src/lib.rs

```diff
@@ -166,7 +168,7 @@ pub struct HandlerCtx<'a> {
     /// The id of the node sending the event
     id: Id,
 
-    c: &'a mut LayoutCtx,
```

Comment:
I'm not sure "layout_ctx" is the best name, though I agree that "c" is terrible. It's an inner layer of state in a somewhat of an onion structure. It also *happens* to overlap 100% at present with the context provided to the `layout` method, but that may not always be true. I might consider a name like "core" (I think "core_state" is an alternative but doesn't really pay for its increased length).